rwengine/src/ai/CharacterController.cpp

```diff
@@ -40,8 +40,10 @@ void CharacterController::skipActivity() {
     // vehicle
     // or jumping.
     if (getCurrentActivity() != nullptr &&
-        getCurrentActivity()->canSkip(character, this))
+        getCurrentActivity()->canSkip(character, this)) {
+        character->activityFinished();
```

Comment:
I'm not sure about the purpose of `activityFinished`.
@danhedron Does it mean: "the activity has finished successfully" or is it more of a [cleanup measure as the comment in the header suggests](https://github.com/rwengine/openrw/blob/79bdd78952d4fdd0071599225ee1807cf9cf4059/rwengine/src/objects/CharacterObject.hpp#L190)

In case it's the later, maybe it should automaticly happen in the destructor or `setActivity` ?

Reply:
It looks like activityFinished flushes actual animation. I think it should be at least renamed.